webapp/components/survey/CategoryDetails/CategoryDetails.js

```diff
@@ -15,35 +15,32 @@ import { useAuthCanEditSurvey } from '@webapp/store/user'
 import { FormItem, Input } from '@webapp/components/form/input'
 import UploadButton from '@webapp/components/form/uploadButton'
 
-import CategoryImportSummary from './components/categoryImportSummary'
-import LevelEdit from './components/levelEdit'
+import ImportSummary from './ImportSummary'
+import LevelEdit from './LevelEdit'
 
-import * as CategoryState from './categoryState'
-import * as Actions from './actions'
+import * as Actions from '../../../loggedin/surveyViews/category/actions'
+import { State, useLocalState } from './store'
+
+const CategoryDetails = (props) => {
+  const { showClose, onCategoryCreated, categoryUuid: categoryUuidProp } = props
 
-const CategoryView = (props) => {
-  const { showClose } = props
   const { categoryUuid } = useParams()
   const i18n = useI18n()
   const dispatch = useDispatch()
   const history = useHistory()
 
-  const category = useSelector(CategoryState.getCategoryForEdit)
-  const importSummary = useSelector(CategoryState.getImportSummary)
   const readOnly = !useAuthCanEditSurvey()
 
+  const { state } = useLocalState({ onCategoryCreated, categoryUuid: categoryUuidProp })
+
+  const category = State.getCategory(state)
+  const importSummary = State.getImportSummary(state)
+
   const validation = Validation.getValidation(category)
   const levels = Category.getLevelsArray(category)
 
   const inCategoriesPath = Boolean(categoryUuid)
```

Comment:
wouldn't useMatchPath more robust?

Reply:
yes, I will handle it in the useInit hook